### PR TITLE
CA-226112: Disable PVS read caching causes System.NullReferenceException

### PR DESCRIPTION
**Pull request by MihaelaStoica**
Signed-off-by: Mihaela Stoica <mihaela.stoica@citrix.com>

### CODE REVIEW

File: XenAdmin/Commands/DisablePvsReadCachingCommand.cs [hunk 1]
```diff
@@ -97,8 +97,7 @@ protected virtual void Execute(IList<VM> vms)
         /// <returns></returns>
         private AsyncAction GetAsyncActionForVm(VM vm)
         {
-            var pvsProxy = vm.Connection.Cache.PVS_proxies.FirstOrDefault(p => p.VM.Equals(vm)); // null if no proxy
-
+            var pvsProxy = vm.PvsProxy;
```
Comment (kc284) [inline XenAdmin/Commands/DisablePvsReadCachingCommand.cs]: Can't we just write in line 67:
```
foreach (var vm in vms)
{
    var pvsProxy = vm.PvsProxy;
    if (pvsProxy != null)
        actions.Add(new PvsProxyDestroyAction(pvsProxy));
}
```
Then we don't need the GetAsyncActionForVm at all - simpler and we gain a few picoseconds by avoiding extra calls and checks.

File: XenAdmin/Dialogs/EnablePvsReadCachingDialog.cs [hunk 2]
```diff
@@ -139,9 +139,7 @@ private AsyncAction GetAsyncActionForVm(VM vm, PVS_site siteSelected)
         /// <returns></returns>
         private bool PvsProxyAlreadyEnabled(VM vm)
```
Comment (kc284) [inline XenAdmin/Dialogs/EnablePvsReadCachingDialog.cs]: Here too I don't think we need to keep this method any longer, we can just keep it simple and write in line 121:
```
if (vm.PvsProxy != null)
    return null;
```

**Comment by MihaelaStoica** [comment]
Thanks @kc284, I've done the changes.

**Comment by kc284** [comment]
Looks ok; merging.

**Pull request merged**